src/Summoner/Source.hs

```diff
@@ -18,26 +21,34 @@ data Source
     = Url Text
     -- | File path to the local source file.
     | File FilePath
+    -- | Link to external file.
+    | Link Text
     deriving (Show, Eq)
 
 sourceT :: Key -> BiToml Source
 sourceT nm = Toml.match (Toml._Text   >>> _Url)  (nm <> "url")
          <|> Toml.match (Toml._String >>> _File) (nm <> "file")
+         <|> Toml.match (Toml._Text >>> _Link) (nm <> "link")
   where
     _Url :: BiMap Text Source
-    _Url = Toml.invert $ Toml.prism (source Just (const Nothing)) Url
+    _Url = Toml.invert $ Toml.prism (source Just (const Nothing) (const Nothing)) Url
 
     _File :: BiMap FilePath Source
-    _File = Toml.invert $ Toml.prism (source (const Nothing) Just) File
+    _File = Toml.invert $ Toml.prism (source (const Nothing) Just (const Nothing)) File
+
+    _Link :: BiMap Text Source
+    _Link = Toml.invert $ Toml.prism (source (const Nothing) (const Nothing) Just) Link
 
-    source :: (Text -> c) -> (FilePath -> c) -> Source -> c
-    source f _ (Url x)  = f x
-    source _ f (File x) = f x
+    source :: (Text -> c) -> (FilePath -> c) -> (Text -> c) -> Source -> c
+    source f _ _ (Url x)  = f x
+    source _ f _ (File x) = f x
+    source _ _ f (Link x) = f x
 
 fetchSource :: Source -> IO (Maybe Text)
 fetchSource = \case
     File path -> catch (Just <$> readFileText path) (fileError path)
     Url url -> catch (fetchUrl url) (urlError url)
+    Link link -> catch (putLink link) (urlError link)
```

Comment:
I guess `catch` is not important in this particular part as the `putLink` function won't produce any error as it's just `pure . Just`.
So you can use just `putLink link` in here.